docs/src/content/en/reference/agents/generate.mdx

```diff
@@ -66,6 +66,38 @@ An optional object that can include configuration for output structure, memory m
       description:
         "Defines the expected structure of the output. Can be a JSON Schema object or a Zod schema.",
     },
+    {
+      name: "memory",
+      type: "object",
+      isOptional: true,
+      description: "Configuration for memory. This is the preferred way to manage memory.",
+      properties: [
+        {
+          parameters: [{
+              name: "thread",
+              type: "string | { id: string; metadata?: Record<string, any>, title?: string }",
+              isOptional: true,
+              description: "The conversation thread, as a string ID or an object with an `id` and optional `metadata`."
+          }]
+        },
+        {
+          parameters: [{
+              name: "resource",
+              type: "string",
+              isOptional: true,
```

Comment:
These can both be required now that they're in a sub object. They were optional before because they were at the top level but if you're using memory you have to pass both of them